design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,187 @@
+This persona would also be expected to define resource classes and other environment specific details.
+* **Application:** An application operator or end user of Crossplane.
+They are focused at a higher level and want to install the functionality of a single application, such as `GitLab` or `WordPress`.
+This persona is not expected to know or control any details about the environments managed by the Crossplane cluster, they simply need to consume their new application of choice.
+
+## Design Summary
+
+The key difference in how the supported personas/scenarios will be handled is in the **scope** of the permissions that they are granted within the control plane.
+
+* **Platform stacks** will be given permissions to watch, access, and create resources at a **cluster level**, i.e. across **all namespaces**.
+* **Application stacks** will be given permissions only within a **single namespace**.
+
+This means that application stacks should not be able to directly affect any resources outside of the specific namespace they are installed into, while platform stacks operate at a higher level of privilege and can affect the entire cluster.
+
+### CRD Scoping
+
+To avoid a common point of confusion, let's review the available scoping of CRDs.
+Custom resource **instances** can be scoped to either a single namespace or to the entire cluster.
+It's important to note that this refers to the instances, not the CRD type definitions.
+From the [Kubernetes documentation](https://kubernetes.io/docs/tasks/access-kubernetes-api/custom-resources/custom-resource-definitions/#create-a-customresourcedefinition):
+
+> CustomResourceDefinitions themselves are non-namespaced and are available to all namespaces
+
+Therefore, any time a CRD is created, it will also be at the cluster scope, even if instances of the CRD are scoped to a namespace.
+This has caused confusion in the past, so it's important to clear that up here.
+
+## Platform Stacks
+
+Platform Stacks can be thought of as a higher level of privilege than other types of stacks, similar to how kernel modules work in Linux.
+The controllers contained in a platform stack will usually reconcile resources that are external to the cluster, for example cloud provider managed services.
+These controllers will also likely be written as full fledged controller-runtime based implementations, but that is not a requirement.
+
+Let's look at a specific example of the GCP platform stack to understand the security and isolation model more thoroughly.
+When the Stack Manager installs the GCP stack, a few things will happen:
+
+* All CRDs related to GCP infrastructure and services will be installed to the cluster, such as `CloudSQLInstance`, `CloudSQLInstanceClass`, `GKECluster`, `GKEClusterClass`, `Network`, `Subnetwork`, etc.
+* A `ClusterRole` will be created at the cluster level that includes all of the RBAC rules for API groups, resources, and verbs that were specified in the `rbac.yaml` file from the stack.
```

Comment:
It may be worth providing a link or some context around what `rbac.yaml` is for uninitiated readers.